geomesa-fs/geomesa-fs-tools/src/main/scala/org/locationtech/geomesa/fs/tools/compact/CompactCommand.scala

```diff
@@ -60,21 +61,21 @@ class CompactCommand extends FsDataStoreCommand with LazyLogging {
       }
       filtered
     }
-    Command.user.info(s"Compacting ${toCompact.size} partitions")
 
     val mode = Option(params.mode).getOrElse {
-      if (PathUtils.isRemote(storage.getMetadata.getRoot.toString)) {
-        RunModes.Distributed
-      } else {
-        RunModes.Local
-      }
+      if (PathUtils.isRemote(storage.getMetadata.getRoot.toString)) { RunModes.Distributed } else { RunModes.Local }
     }
 
+    Command.user.info(s"Compacting ${toCompact.size} partitions in ${mode.toString.toLowerCase(Locale.US)} mode")
+
+    // compact metadata up-front, that will save us listing redundant metadata files for each partition
+    storage.getMetadata.compact()
+
     mode match {
       case RunModes.Local =>
-        toCompact.map(_.name).foreach { p =>
-          logger.info(s"Compacting $p")
-          storage.compact(p)
+        toCompact.par.foreach { p =>
```

Comment:
Do we have control over the number of threads used by this .par?

Reply:
you can control it through various scala-level settings - it defaults to numProcessors I believe. But I've cleaned this up in a follow-on PR that exposes num-threads as a parameter: https://github.com/locationtech/geomesa/pull/2176/files#diff-acb9f51edcb091704a605d5517f01b45R140